plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/writer/GcsTransactionLogSynchronizer.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.transactionlog.writer;
+
+import com.google.api.client.http.InputStreamContent;
+import com.google.api.client.util.Data;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.model.StorageObject;
+import com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem;
+import com.google.cloud.hadoop.gcsio.CreateFileOptions;
+import com.google.cloud.hadoop.gcsio.CreateObjectOptions;
+import com.google.cloud.hadoop.gcsio.PathCodec;
+import com.google.cloud.hadoop.gcsio.StorageResourceId;
+import com.google.common.collect.Maps;
+import io.trino.hdfs.HdfsContext;
+import io.trino.hdfs.HdfsEnvironment;
+import io.trino.plugin.deltalake.GcsStorageFactory;
+import io.trino.spi.connector.ConnectorSession;
+import org.apache.hadoop.fs.Path;
+
+import javax.inject.Inject;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.util.Base64;
+import java.util.Map;
+
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageFileSystem.objectOptionsFromFileOptions;
+import static io.trino.plugin.hive.util.HiveWriteUtils.getRawFileSystem;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class GcsTransactionLogSynchronizer
+        implements TransactionLogSynchronizer
+{
+    private static final Base64.Encoder base64Encoder = Base64.getEncoder();
+
+    private final GcsStorageFactory gcsStorageFactory;
+    private final HdfsEnvironment hdfsEnvironment;
+
+    @Inject
+    public GcsTransactionLogSynchronizer(HdfsEnvironment hdfsEnvironment, GcsStorageFactory gcsStorageFactory)
+    {
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.gcsStorageFactory = requireNonNull(gcsStorageFactory, "gcsStorageFactory is null");
+    }
+
+    // This approach should be compatible with OSS Delta Lake.
```

Comment:
Yes, it "should be". Is it?
Either remove the comment, or replace with one stating the fact.
You can also link to a specific file in Delta project that implements same logic there.

Reply:
Removing the comment
This is a left-over from the initial draft which copied partly code from  `AzureTransactionLogSynchronizer`